ggshield/core/git_hooks/ci/previous_commit.py

```diff
@@ -0,0 +1,163 @@
+    # because this one is only populated in case of PR
+    # whereas push_before_sha can be populated in PR in case of
+    # push force event in a PR
+    if pull_req_base_sha:
+        return pull_req_base_sha
+
+    if push_before_sha:
+        return push_before_sha
+
+    raise UnexpectedError(
+        "Unable to get previous commit. Please submit an issue with the following info:\n"
+        "  Repository URL: <Fill if public>\n"
+        f"github_push_before_sha: {push_before_sha}\n"
+        f"github_pull_base_sha: {pull_req_base_sha}\n"
+    )
+
+
+def github_push_previous_commit_sha() -> Optional[str]:
+    return os.getenv("GITHUB_PUSH_BEFORE_SHA")
+
+
+def github_pull_request_previous_commit_sha() -> Optional[str]:
+    targeted_branch = os.getenv("GITHUB_BASE_REF")
+
+    # Not in a pull request workflow
+    if targeted_branch is None:
+        return None
+
+    # The branch is not directly available in CI env
+    # We need to get commits through remotes
+    last_target_commit = get_list_commit_SHA(
+        f"remotes/origin/{targeted_branch}", max_count=1
+    )
+
+    # Unable to find a commit on this branch
+    # Consider it empty
+    if len(last_target_commit) == 0:
+        return None
+
+    return last_target_commit[0]
+
+
+def gitlav_previous_commit_sha(verbose: bool) -> str:
+    push_before_sha = gitlab_push_previous_commit_sha()
+    merge_req_base_sha = gitlab_merge_request_previous_commit_sha()
+
+    if verbose:
+        click.echo(
+            f"gitlab_push_before_sha: {push_before_sha}\n"
+            f"gitlab_merge_base_sha: {merge_req_base_sha}\n",
+            err=True,
+        )
+
+    # Always EMPTY_SHA in MR pipeline according with
+    # https://docs.gitlab.com/ee/ci/variables/predefined_variables.html
+    if push_before_sha and push_before_sha is not EMPTY_SHA:
+        return push_before_sha
+
+    if merge_req_base_sha:
+        return merge_req_base_sha
+
+    raise UnexpectedError(
+        "Unable to get previous commit. Please submit an issue with the following info:\n"
+        "  Repository URL: <Fill if public>\n"
+        f"gitlab_push_before_sha: {push_before_sha}\n"
+        f"gitlab_pull_base_sha: {merge_req_base_sha}\n"
+    )
+
+
+def gitlab_push_previous_commit_sha() -> Optional[str]:
+    return os.getenv("CI_COMMIT_BEFORE_SHA")
+
+
+def gitlab_merge_request_previous_commit_sha() -> Optional[str]:
+    targeted_branch = os.getenv("CI_MERGE_REQUEST_TARGET_BRANCH_NAME")
+
+    # Not in a pull request workflow
+    if targeted_branch is None:
+        return None
+
+    # The branch is not directly available in CI env
+    # We need to get commits through remotes
+    last_target_commit = get_list_commit_SHA(
+        f"remotes/origin/{targeted_branch}", max_count=1
+    )
+
+    # Unable to find a commit on this branch
+    # Consider it empty
+    if len(last_target_commit) == 0:
+        return None
+
+    return last_target_commit[0]
```

Comment:
this is basically the same method as for github, the env var is different, can you maybe factorize it ? at least this part